cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -493,17 +487,21 @@ private void checkReconfigurability(PodRef podRef, Pod pod, RestartContext resta
                 needsReconfig = true;
             }
         } else if (needsRestart) {
-            LOGGER.infoCr(reconciliation, "Pod {} needs to be restarted. Reason: {}", podRef, reasonToRestartPod);
+            LOGGER.infoCr(reconciliation, "Pod {} needs to be restarted. Reason: {}", podRef, reasonsToRestartPod.getAllReasonNotes());
         }
 
         if (podStuck)   {
             LOGGER.infoCr(reconciliation, "Pod {} needs to be restarted, because it seems to be stuck and restart might help", podRef);
+            reasonsToRestartPod.add(RestartReason.POD_STUCK);
         }
-        restartContext.needsRestart = needsRestart;
-        restartContext.needsReconfig = needsReconfig;
-        restartContext.forceRestart = podStuck;
-        restartContext.diff = diff;
-        restartContext.logDiff = loggingDiff;
+
+        kafkaRestartContext.setRestartReasons(reasonsToRestartPod);
```

Comment:
Is there a larger semantically meaningful action that these could be collapsed into?